### PR TITLE
[#11020] Remove searching response comments related features and search box at home page

### LINKED ISSUE #3918: Instructor: search: Allow searching feedback responses, questions
Currently, it allows searching for feedback response comments but not feedback responses.

### LINKED ISSUE #8704: Keep comment search documents up-to-date
Root cause of #8702 - see https://github.com/TEAMMATES/teammates/pull/8698#issuecomment-375829424. Results in outdated search results.

The only time comment search documents are ever updated/deleted is when comments are individually edited/deleted, and this is done at the action level! Updating/deletion of responses, questions, sessions and courses does not result in updating/deletion of corresponding comment search documents, resulting in lots of outdated/orphaned documents, leading to outdated search results.

Updates to documents should be done at the logic level and should follow updates to responses, questions, sessions and courses closely.

### LINKED ISSUE #9103: Searching: Same session name in different courses will produce malformed results
- **Environment**: `master`

**Steps to reproduce**

- Create two different courses.
- Create one session in two different courses with the same name
- Submit some feedback responses as students in both sessions
- Add a comment with the same content as instructors to feedback response in different sessions
- Go to `instructorSearchPage` and search for the comment you just entered.

search instructor

**Expected behaviour**

- Results should display in two separated sessions.

**Actual behaviour**

- Results displayed in one session.

**Additional info**

https://github.com/TEAMMATES/teammates/blob/290443f86eaf741c9fda0670070b8fd6c269ad90/src/main/java/teammates/storage/search/FeedbackResponseCommentSearchDocument.java#L303-L324

Session name is not unique in the whole DB. Should use (session name + courseID)

### PR DESCRIPTION
**Pull request by wkurniawan07**
First part of #11020 

This removes all code related to searching response comments as well as the search box at the top of instructor home page.

This will also close #3918, close #8704, close #9103 since they are all irrelevant without the searching response comments feature.

### CODE REVIEW

File: src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.html
Comment (Derek-Hardy) [inline src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.html:10]: Is there a good reason to remove the home search bar?

Understand it's just a redirection to the search page. From the UX perspective, this removal makes search operation become two-steps-away from login.

If search is a frequent action performed by the instructor, it will be convenient to make it immediately available at home page?

**Review by Derek-Hardy — APPROVED** [review]
LGTM!

File: src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.html
Comment (wkurniawan07) [reply src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.html:10]: A search bar at home page heavily implies a site-wide search, which this search bar is nothing about. There are multiple cases of such mis-usage, like "how do I publish a session", "how do I enroll students", etc.

Comment (jianhandev) [reply src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.html:10]: Agree that the search bar was a bit out of place on the homepage, esp. since we already have a search page. However, I still think that the current search bar is not explicit enough as to what is searchable and what is not. I believe we can address that in a new issue or after the search migration.

**Review by jianhandev — COMMENTED** [review]
There are still some mention of search comments feature in the following files:
- `features-page.component.html` (line 212)
- `instructor-search-page.component.ts` (line 42)

File: src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.html
Comment (wkurniawan07) [reply src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.html:10]: Screenshot 2021-03-26 at 6 45 30 PM

**Comment by wkurniawan07** [comment]
@jianhandev I will merge after your approval

**Review by jianhandev — APPROVED** [review]
LGTM!

**Pull request merged**
